source/taskmanager-client/src/main/java/nl/aerius/taskmanager/client/util/QueueHelper.java

```diff
@@ -66,8 +70,11 @@ public static byte[] objectToBytes(final Serializable object) throws IOException
     if (object == null) {
       bytes = new byte[0];
     } else {
-      try (final ByteArrayOutputStream bo = new ByteArrayOutputStream(); final ObjectOutputStream out = new ObjectOutputStream(bo)) {
-        out.writeObject(object);
+      try (final ByteArrayOutputStream bo = new ByteArrayOutputStream()) {
+        try (final GZIPOutputStream gzip = new GZIPOutputStream(bo);
```

Comment:
Please add a comment that this is needed because the `GZIPOutputStream` is rather lazy. Otherwise we might refactor this and introduce a bug.